### ISSUE TITLE
Data resource for Dedicated Server

### ISSUE THREAD

**Issue by phumberdroz** [mwudka/terraform-provider-hetznerrobot#2]
It would be nice to have a data resource for dedicated servers

**Comment by mwudka** [comment]
Agreed, that's a great idea. I hope to have time to take a stab at this later in the week. Are there any specific filters you'd like to have on the data source? I was thinking name and number to start, but open to suggestions!

**Comment by SLoeuillet** [comment]
Hello

Started a server/boot datasource/resource in forked repo
Still in WIP

**Comment by mwudka** [comment]
Very cool! I look forward to reviewing + merging when you are ready!

**Comment by SLoeuillet** [comment]
That's my first time with Go but having GoLand as IDE helped a lot 

Could you share your terraform.rc config used to test the plugin locally ?
I have trouble with the "Provider not signed" part and local version enumeration.

You can already take a look at the current code in my SLoeuillet github organisation.

Cheers

**Comment by mwudka** [comment]
Yes, GoLand is indeed a wonderful help.

For testing locally, I run `make install` from the repo. This installs the plugin into `~/.terraform.d/plugins/github.com/mwudka/hetznerrobot/.....`. After doing that, in your test terraform config include a block like this:

```
terraform {
  required_providers {
    hetznerrobot = {
      version = "0.0.1"
      source  = "github.com/mwudka/hetznerrobot"
    }
  }
}
```

That will tell Terraform to get that provider from your locally installed path, rather than reaching into the registry to pull it. 

It looks like you have many positive changes in the works; I'm looking forward to reviewing your PR when it's ready! I did notice that you removed the vendor directory. I generally like to keep the vendor directory checked in, because that way the builds are a bit more reproducible and we can easily navigate through all the dependencies when figuring things out. It also provides a nice backstop in case a dependency every vanishes from the internet. However, it is open for discussion as with all things!

I also recommend you check out the neat things @remilapeyre is working on in #1, specifically the move to factor out the Hetzner robot API calls into their own client.

**Comment by SLoeuillet** [comment]
@mwudka 

Got this when using your plugin localy :

Partner and community providers are signed by their developers.
If you'd like to know more about provider signing, you can read about it here:
https://www.terraform.io/docs/cli/plugins/signing.html
╷
│ Error: Invalid provider registry host
│ 
│ The given source address "github.com/mwudka/hetznerrobot" specifies a
│ GitHub repository rather than a Terraform provider. Refer to the
│ documentation of the provider to find the correct source address to use.

**Comment by SLoeuillet** [comment]
Created a PR with all my implemented resources/datasources (PR3)
